ui/pages/confirmations/components/confirm/info/shared/redesigned-gas-fees/redesigned-gas-fees.tsx

```diff
@@ -0,0 +1,333 @@
+import { TransactionMeta } from '@metamask/transaction-controller';
+import React, { useEffect, useMemo, useState } from 'react';
+import { useSelector } from 'react-redux';
+import { EtherDenomination } from '../../../../../../../../shared/constants/common';
+import { EditGasModes } from '../../../../../../../../shared/constants/gas';
+import {
+  addHexes,
+  getEthConversionFromWeiHex,
+  getValueFromWeiHex,
+  subtractHexes,
+} from '../../../../../../../../shared/modules/conversion.utils';
+import { getMinimumGasTotalInHexWei } from '../../../../../../../../shared/modules/gas.utils';
+import {
+  ConfirmInfoRow,
+  ConfirmInfoRowVariant,
+} from '../../../../../../../components/app/confirm/info/row';
+import {
+  Box,
+  Button,
+  ButtonSize,
+  ButtonVariant,
+  IconName,
+  Text,
+} from '../../../../../../../components/component-library';
+import { getConversionRate } from '../../../../../../../ducks/metamask/metamask';
+import {
+  AlignItems,
+  Display,
+  FlexDirection,
+  IconColor,
+  JustifyContent,
+  TextAlign,
+  TextColor,
+  TextVariant,
+} from '../../../../../../../helpers/constants/design-system';
+import { useFiatFormatter } from '../../../../../../../hooks/useFiatFormatter';
+import {
+  currentConfirmationSelector,
+  getCurrentCurrency,
+} from '../../../../../../../selectors';
+import AdvancedGasFeePopover from '../../../../advanced-gas-fee-popover';
+import EditGasFeePopover from '../../../../edit-gas-fee-popover';
+import EditGasPopover from '../../../../edit-gas-popover';
+import GasTiming from '../../../../gas-timing';
+import { useSupportsEIP1559 } from '../../hooks/supports-eip-1559';
+import { useEIP1559TxFees } from '../../hooks/use-eip-1559-fees';
+import { EditGasIcon } from '../edit-gas-icon/edit-gas-icon';
+
+export const RedesignedGasFees = () => {
+  const currentConfirmation = useSelector(
+    currentConfirmationSelector,
+  ) as TransactionMeta;
+
+  if (!currentConfirmation?.txParams) {
+    return null;
+  }
+
+  const [showCustomizeGasPopover, setShowCustomizeGasPopover] = useState(false);
+  const closeCustomizeGasPopover = () => setShowCustomizeGasPopover(false);
+  const { supportsEIP1559 } = useSupportsEIP1559(currentConfirmation);
+
+  let gasEstimate;
+  if (supportsEIP1559) {
+    const baseFeePerGas = subtractHexes(
+      currentConfirmation.txParams.maxFeePerGas as string,
```

Comment:
Given we have many references to the params, should we destructure what we need at the start of the function to avoid the repeated property chains?